pkg/api/alerting/v2alpha1/types.go

```diff
@@ -442,3 +449,79 @@ func parseLabelFilters(req *restful.Request) (map[string]string, map[string]stri
 	}
 	return labelEqualFilters, labelContainFilters
 }
+
+const (
+	ErrBadData       ErrorType = "bad_data"
+	ErrDuplicateName ErrorType = "duplicate_name"
+	ErrNotFound      ErrorType = "not_found"
+	ErrServer        ErrorType = "server_error"
+
+	StatusSuccess Status = "success"
+	StatusError   Status = "error"
+
+	ResultCreated Result = "created"
+	ResultUpdated Result = "updated"
+	ResultDeleted Result = "deleted"
+)
+
+type Status string
+
+type ErrorType string
+
+type Result string
+
+type BulkResponse struct {
+	Errors bool                `json:"errors" description:"If true, one or more of the operations in the bulk request did not complete successfully"`
+	Items  []*BulkItemResponse `json:"items" description:"It contains the result of each operation in the bulk request"`
+}
+
+func (br *BulkResponse) Derive() *BulkResponse {
```

Comment:
Considering use a different name for this function, Derive means subclass in C++